### PR TITLE
Remove trivial guards

### PR DESCRIPTION
**Pull request by larskuhtz**
This PR removes outdated kill switch logic and outdated guards that will always evaluate to the same value.

Removing this redundant logic makes room for new guards related to the upcoming forking changes (creation time fix, hash function changes, 20 chains).

Some of the removed code was already marked for removal.

### CODE REVIEW

File: src/Chainweb/Mempool/InMem.hs [hunk 1]
```diff
@@ -162,10 +161,10 @@ withInMemoryMempool_ :: Logger logger
                      -> ChainwebVersion
                      -> (MempoolBackend t -> IO a)
                      -> IO a
-withInMemoryMempool_ l cfg v f = do
+withInMemoryMempool_ l cfg _v f = do
```
Comment (emilypi) [inline src/Chainweb/Mempool/InMem.hs]: why not just remove the version from the signature? Same for below

File: src/Chainweb/Pact/TransactionExec.hs [hunk 2]
```diff
@@ -145,19 +142,16 @@ applyCmd
...
   where
     txst = TransactionState mcache0 mempty 0 Nothing (_geGasModel freeGasEnv)
 
-    executionConfigNoHistory = mkExecutionConfig $
-      [ FlagDisableHistoryInTransactionalMode ] ++
-      [ FlagDisableModuleInstall | not ecMod ] ++
-      [ FlagOldReadOnlyBehavior | isPactBackCompatV16 ]
+    executionConfigNoHistory = mkExecutionConfig
+      $ FlagDisableHistoryInTransactionalMode
+      : [ FlagOldReadOnlyBehavior | isPactBackCompatV16 ]
```
Comment (emilypi) [inline src/Chainweb/Pact/TransactionExec.hs]: is this... a list comprehension? I've never seen that style before!

File: test/Chainweb/Test/Pact/TransactionTests.hs [hunk 3]
```diff
@@ -226,17 +226,23 @@ testCoinbase797DateFix = testCaseSteps "testCoinbase791Fix" $ \step -> do
...
+    postForkHeight = 121452
+
     logger = newLogger neverLog ""
 
+    -- | someBlockHeader is a bit slow for the vuln797Fix to trigger. So, instead
+    -- of mining a full chain we fake the height.
+    --
+    mkTestHeader :: BlockHeight -> BlockHeader
+    mkTestHeader h = (someBlockHeader (FastTimedCPM singleton) 10)
+        { _blockHeight = h }
```
Comment (emilypi) [inline test/Chainweb/Test/Pact/TransactionTests.hs]: 👍

File: src/Chainweb/Mempool/InMem.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/Mempool/InMem.hs]: This is part of the main mempool API, which I did't want to change. It also seems reasonable that the mempool is aware of the chainweb version, even if the current implementation doesn't use it.

File: src/Chainweb/Pact/TransactionExec.hs (hunk 2 above)
Comment (larskuhtz) [reply src/Chainweb/Pact/TransactionExec.hs]: yes, list comprehension are desugared to `MonadPlus` computations. A boolean expression `exp` becomes `guard exp`. It's quite useful to filter the results of the comprehension.

Comment (slpopejoy) [reply src/Chainweb/Pact/TransactionExec.hs]: I switched to comprehensions in a recent PR, but the credit goes entirely to Dr Kuhtz (used in block header validation)

**Pull request merged**